From 9074f99f1684f70ccd26a85d8d08776dfbf188db Mon Sep 17 00:00:00 2001
From: Ralph Sennhauser <ralph.sennhauser@gmail.com>
Date: Sun, 17 Aug 2025 20:05:47 +0200
Subject: [PATCH] Do not require literal message ids in Atlas

In >=wxWidgets-3.3 there is a new static check for literal message
ids[1] so that xgettext would reliable work to extract strings to
translate. As we don't translate Atlas at all nor use xgettext just
disable the check to allow building against >=wxWidgets-3.3.

[1] https://wxwidgets.org/help/msgid-literals/

Signed-off-by: Ralph Sennhauser <ralph.sennhauser@gmail.com>
---
 build/premake/extern_libs5.lua | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/premake/extern_libs5.lua b/build/premake/extern_libs5.lua
index 4f51c2c0ed..f3de8e0ec9 100644
--- a/build/premake/extern_libs5.lua
+++ b/build/premake/extern_libs5.lua
@@ -734,6 +734,7 @@ extern_lib_defs = {
 				-- so we have to use wxwidgets' own config tool
 				pkgconfig.add_includes(nil, wx_config_path(), "--unicode=yes --cxxflags")
 			end
+			defines({ "wxNO_REQUIRE_LITERAL_MSGIDS" })
 		end,
 		link_settings = function()
 			if os.istarget("windows") then
-- 
2.47.3

